MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+2. Relayer awaits for finalization of the transaction on L1.
+3. Relayer completes the bridge on L2.
+4. Completion transaction verifies that the transaction is truthful by comparing the provided bridgeTransferId hash and the emitted values of originator, recipient, amount and nonce. Finally, it mints MOVE to the recipient address.
+5. User is notified on the frontend that their transaction has been completed.
+6. In case of a rare case of a reorg on the recipient network that leads to the transaction removal from the user perspective, the transaction can be redone and relayer multisig can handle it.
+
+![L1-L2](l1l2.png)
+
+L2 -> L1
+
+1. User initiates a bridge transfer on L2. Contract stores a mapping of the user bridgeTransferIds for easy access. It transfers from the user MOVE amount to the contract and burns it. Transaction emits originator, recipient, amount and nonce.
+2. Relayer awaits for finalization of the transaction on L2.
+3. Relayer completes the bridge on L1.
+4. Completion transaction verifies that the transaction is truthful by comparing the provided bridgeTransferId hash and the emitted values of originator, recipient, amount and nonce. Finally, it transfers MOVE to the recipient address, which has been previously bridged from L1.
+5. User is notified on the frontend that their transaction has been completed.
+6. In case of a rare caase of a reorg on the recipient network that leads to the transaction removal from the user perspective, the transaction can be redone and relayer multisig can handle it.
+
+![L2-L1](l2l1.png)
+
+### Key Features
+
+1. **Two-Transaction Model**:
+   - **Initiation**: User sends a transaction to initiate the bridge containing recipient and amount.
+   - **Completion**: A relayer or multi-signature group completes the transfer on the counterparty contract with the originator, recipient, amount and noce for hash verification.
+   - **No Funds Requirement**: User is not required to have funds on receiving chain and we do not have to build sponsored transactions.
+   - **Delivery by Relayer**: Because the relayer delivers the funds, there is no requirement for the user to complete the transaction on receiving chain, therefore massively simplifying the user experience and allowing a smooth onboarding on the network for bridges from L1 to L2. User only has to await for the finalization on the source chain and for the relayer to perform the completion on receiving chain. This is standard practice for every major bridge in the ecosystem.
+   - **Less parameters**: Because there is no exchange of secrets between the user and relayer, we have a substantial reduction of logic.
+   - **Only Completable**: Currently we reserve a refunder role to revert transactions. This approach is different where we guarantee delivery of funds through the same party that would guarantee funds being refunded, because bridges can ONLY be completed.
```

Comment:
I don't understand

> because bridges can ONLY be completed.

Reply:
There is no refund, it can only be completed by the Relayer.